### PR TITLE
issue #2158 pgp back to bg script

### LINKED ISSUE #2158: prep work for manifest V3 - move pgp functionality from background script to content script
First should do some prep work - move what we can from background script to content script

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-browser/issues/1509#issuecomment-548742536_

### PR DESCRIPTION
**Pull request by limonte**
Closes #2158

### CODE REVIEW

**Comment by tomholub** [comment]
I'd also look at removing all `parentTabId` from https://github.com/FlowCrypt/flowcrypt-browser/blob/master/test/tests.json so that they don't cause confusion

File: test/tests.json [hunk 1]
```diff
@@ -16,14 +16,14 @@
...
...GbsVcBxuQigilyMpDQJJfcUiqfN8KL%2BN8ICpnuPGgaMQ97SLeHq2Mmm%0AehcEZwVQZGlCnQJNKmhbqqxJB7WmdBRKTDiBxE5qz5r3grB%2F5v%2BMbyM6G6MyAnkP%0AA%2FUKX0QUZsPDR41XVWhZPTDo%2F%2FZ6aIKJwlgB3E9vak2JkD4%2FpdgzyAM28HOTUyJ%2F%0ASfBVLd%2B%2FjxHIVlm1IaLUAzvJjG0NFlXvD7Pkzs5pXmUf%2F%2FbTdFXNA3uh7VBSGQMl%0AkaeyuemQwiW2Ray4tYbUq%2FFCzl%2B8862JBY98w38natrA%2B%2BWMLHIox0rhMIG%2FvoyK%0AU1%2B2KKgED414MsC309jn%2BP6WCZGKt34BXSfDp%2FRbgwP3X0QIxSYOxtmX8fjKlVPR%0A9FPmkwFvIYsE14MSB16y2vxSEt8JFKoGhXRuVxlGoYuuZrpERfhynnkwSLkw0zln%0AMNUUihw9AePivi6H0qy%2B7DpUy%2B41CW8nxkx%2FdePcdbAq84Y71FyfM7gbLu04EPZ0%0AhTzzgSLDSWvLc7vWRGbqI1erQhfadQwiUzMd0YAyImWmnm003dxfRNNC0oCYDE8V%0A3QGFqOmr4AqcbMGIWBGiP0LajmehJEv%2B8GkIkuDwQRtkgaAkHwDMigujFtraqGEn%0AmduYmYBW88YDsXD9Jv24d4Pt2Ce7P4lc4DEAU3vqUMZFdIwHanjKSNr8O6aXXd0e%0AwrFjc71tUTNGF0suNi%2B74ol0rlS1seQNiijulEW53ngK8z5brNSd1N56H%2FwcYx81%0AqSyeqnGYHphNbpwhBTqHkURBlwygxI2%2BCuMSR96j49ko9AZZHYl6DBAqHLKYWHGi%0AvqvpG%2F%2B%2BFFy0AMSxjs%2F%2FCe5lTS3y3skfneaZ9sgH7o%2BUXCJ7Op%2BmtLdHGjn6%0A%3DGAOS%0A-----END%20PGP%20MESSAGE-----&hasPassword=___cu_false___&msgId=16f66f1da9d50d05&senderEmail=michael%20<censored%40email.com>&isOutgoing=___cu_false___&acctEmail=flowcrypt.compatibility%40gmail.com"
```
Comment (tomholub) [inline test/tests.json]: There's still a bunch more in this file (every one had a partentTabId - I think without it the frame would show an error). They may be all sorts of strings, I'd remove them all:

image

**Comment by tomholub** [comment]
GitHub is telling me there's a comment but don't see any, just want to make sure I didn't miss any question :)

image

**Comment by limonte** [comment]
> GitHub is telling me there's a comment but don't see any, just want to make sure I didn't miss any question :)
> 
> image

GitHub will notify about both new commits and new comments in a PR. To ensure there are no unanswered conversations, I'll ping you on conversations waiting for your attention after 24h of inactivity. Would that work for you?

File: test/tests.json (hunk 1 above)
Comment (limonte) [reply test/tests.json]: Done :+1:

**Comment by limonte** [comment]
The last thing I stuck in this PR is `Error: BrowserMsg.sendAwait(pgpMsgDecrypt) returned(null) with lastError: (no lastError)` in this case:

```
/ci_pgp_host_page.htm?account_email=flowcrypt.compatibility%40gmail.com&frame_id=frame_yVMKFLRDiY&message=&has_password=___cu_false___&message_id=1622eaa286f90737&sender_email=human%40flowcrypt.com&is_outgoing=___cu_false___
```

Looks like `const decrypted = await prepared.message.decrypt(privateKeys, passwords, undefined, false);` just throwing an undefined exception and I can't figura out why. Could you help me with this one?

**Comment by tomholub** [comment]
> > GitHub is telling me there's a comment but don't see any, just want to make sure I didn't miss any question :)
> > image
> 
> GitHub will notify about both new commits and new comments in a PR. To ensure there are no unanswered conversations, I'll ping you on conversations waiting for your attention after 24h of inactivity. Would that work for you?

Definitely

**Comment by tomholub** [comment]
> /ci_pgp_host_page.htm?account_email=flowcrypt.compatibility%40gmail.com&frame_id=frame_yVMKFLRDiY&message=&has_password=___cu_false___&message_id=1622eaa286f90737&sender_email=human%40flowcrypt.com&is_outgoing=___cu_false___

I'm able to reproduce, and will have a look at this tomorrow.

chrome-extension://nhlmpmiobabndkmnhocbgomijbcmiikg/chrome/dev/ci_pgp_host_page.htm?account_email=flowcrypt.compatibility%40gmail.com&frame_id=frame_yVMKFLRDiY&message=&has_password=___cu_false___&message_id=1622eaa286f90737&sender_email=human%40flowcrypt.com&is_outgoing=___cu_false___

**Comment by tomholub** [comment]
The source of the issue is that the browser keeps sending the messages to both background and the parent tab, but I'm not yet sure how to fix it, tomorrow I'll figure something out.

**Comment by limonte** [comment]
> The source of the issue is that the browser keeps sending the messages to both background and the parent tab, but I'm not yet sure how to fix it, tomorrow I'll figure something out.

Not sure if it's related or not, but with changes from this PR I'm facing another issue in Firefox only:

> TypeError: openpgp is undefined

when trying to open any encrypted mail from myself. The issue is happening only in Firefox.

**Comment by tomholub** [comment]
Fixed, but only on Chrome. I'll take a look at Firefox

**Comment by tomholub** [comment]
> > The source of the issue is that the browser keeps sending the messages to both background and the parent tab, but I'm not yet sure how to fix it, tomorrow I'll figure something out.
> 
> Not sure if it's related or not, but with changes from this PR I'm facing another issue in Firefox only:
> 
> > Message is either badly formatted or not compatible with FlowCrypt.
> > TypeError: openpgp is undefined
> 
> when trying to open any encrypted mail from myself. The issue is happening only in Firefox.

Indeed, on Firefox, openpgp.js seems to be undefined in content script. I don't know why, will try a few more things.

**Comment by limonte** [comment]
oops, pushed the commit to the wrong branch, sorry!

**Comment by tomholub** [comment]
I seem to be running into a wall, OpenPGP.js really cannot be loaded in Firefox content script. That's indeed quite a twist to this set of PRs. I may need to raise this with the OpenPGP.js team.

**Comment by tomholub** [comment]
I'll keep all of the changes that were refactored in these PRs (so that it's easier to switch later) but will start sending the requests back to bg script as before.. until the Firefox situation is understood and resolved. 

Sorry that there were so many dragons in this issue, it was completely unexpected.

I'll wrap this up.

**Pull request merged**